### PR TITLE
[build] support CENTEC SAI 1.0 on 201712 branch and update e582-48x6q board  

### PR DESCRIPTION
**Pull request by chenyq2**
Signed-off-by: chenyq chenyq@centecnetworks.com

    What I did
    update CENTEC platform to support SAI1.0 on 201712 branch;
    support 40G port for e582-48x6q board;
    disable advanced function on e582-48x6q by swssconfig.sh;
    set acpi_enforce_resources=no to for sdk dal.ko;
    disable thrift for compile issue; 

    How to verify it
    build centec board image and test SAI1.0 about function on e582-48x6q board.

### CODE REVIEW

File: dockers/docker-orchagent/swssconfig.sh [hunk 1]
```diff
@@ -29,6 +29,8 @@ HWSKU=`sonic-cfggen -d -v "DEVICE_METADATA['localhost']['hwsku']"`
 
 SWSSCONFIG_ARGS="00-copp.config.json ipinip.json ports.json switch.json "
 
+if [ "$HWSKU" == "E582-48x6q" ]; then
+    SWSSCONFIG_ARGS="" 
 if [ "$HWSKU" == "Force10-S6000" ]; then
```
Comment (lguohan) [inline dockers/docker-orchagent/swssconfig.sh]: please change this to elif

File: platform/centec/libsaithrift-dev.mk [hunk 2]
```diff
@@ -4,4 +4,4 @@ LIBSAITHRIFT_DEV_CENTEC = libsaithrift-dev_0.9.4_amd64.deb
 $(LIBSAITHRIFT_DEV_CENTEC)_SRC_PATH = $(SRC_PATH)/SAI
 $(LIBSAITHRIFT_DEV_CENTEC)_DEPENDS += $(LIBTHRIFT) $(LIBTHRIFT_DEV) $(THRIFT_COMPILER) $(CENTEC_SAI)
 $(LIBSAITHRIFT_DEV_CENTEC)_RDEPENDS += $(LIBTHRIFT) $(CENTEC_SAI)
-SONIC_DPKG_DEBS += $(LIBSAITHRIFT_DEV_CENTEC)
+#SONIC_DPKG_DEBS += $(LIBSAITHRIFT_DEV_CENTEC)
```
Comment (lguohan) [inline platform/centec/libsaithrift-dev.mk]: no need to comment this line.

File: platform/centec/rules.mk [hunk 3]
```diff
@@ -8,12 +8,10 @@ include $(PLATFORM_PATH)/libsaithrift-dev.mk
 include $(PLATFORM_PATH)/python-saithrift.mk
 include $(PLATFORM_PATH)/docker-ptf-centec.mk
 
-SONIC_ALL += $(SONIC_ONE_IMAGE) \
-             $(DOCKER_PTF_CENTEC) \
-             $(DOCKER_SYNCD_CENTEC_RPC)
```
Comment (lguohan) [inline platform/centec/rules.mk]: why comment out DOCKER_PTF_CENTEC and DOCKER_SYNCD_CENTEC_RPC?

File: dockers/docker-orchagent/swssconfig.sh (hunk 1 above)
Comment (chenyq2) [reply dockers/docker-orchagent/swssconfig.sh]: accept

File: platform/centec/libsaithrift-dev.mk (hunk 2 above)
Comment (chenyq2) [reply platform/centec/libsaithrift-dev.mk]: accept

Comment (chenyq2) [reply platform/centec/libsaithrift-dev.mk]: my compile is failed caused by this line. the log said that there are no src/SAI derectory for pushd.
 and referenced from BCM, Mellenox who can pass build on master, i comment it.

File: platform/centec/rules.mk (hunk 3 above)
Comment (chenyq2) [reply platform/centec/rules.mk]: DOCKER_PTF_CENTEC and DOCKER_SYNCD_CENTEC_RPC depend on LIBSAITHRIFT_DEV_CENTEC

**Comment by chenyq2** [comment]
currently all BCM, MNX, NEPHOS, P4 comment "SONIC_DPKG_DEBS += $(LIBSAITHRIFT_DEV_XXX)" . it seems they all have the same iuuse from libsaithrift-dev.mk. 
in fact, i don't know how to get or update "libsaithrift-dev_0.9.4_amd64.deb". please help finger out how to fix it.

File: dockers/docker-orchagent/swssconfig.sh [hunk 4]
```diff
@@ -29,7 +29,9 @@ HWSKU=`sonic-cfggen -d -v "DEVICE_METADATA['localhost']['hwsku']"`
 
 SWSSCONFIG_ARGS="00-copp.config.json ipinip.json ports.json switch.json "
 
-if [ "$HWSKU" == "Force10-S6000" ]; then
+if [ "$HWSKU" == "E582-48x6q" ]; then
+    SWSSCONFIG_ARGS=""
```
Comment (lguohan) [inline dockers/docker-orchagent/swssconfig.sh]: on centec platform, I assume you need to apply copp.config.json, ipinip.json, ports.json as well as switch.json.

Comment (chenyq2) [reply dockers/docker-orchagent/swssconfig.sh]: yes, that's OK.

**Pull request merged**